### DISCUSSION TITLE
"please convert to new-style JPEG compression and notify vendor of writing software"

### DISCUSSION THREAD

**Discussion by P2AK** [ImageMagick/ImageMagick#5000] [Help] [answered]
I'm trying to composite one image onto another but keep getting this error message about JPEG compression.

This is the command I'm trying to use:
`convert gravesonly-2.tif 001.TIF -compose over -composite test.tif`

This is the resulting error:
`convert: Deprecated and troublesome old-style JPEG compression mode, please convert to new-style JPEG compression and notify vendor of writing software. OJPEGSetupDecode @ warning/tiff.c/TIFFWarnings/964.`

Is there a conversion command I can run on either of the offending files in order to resolve this?

The two files in question can be found [here](https://filebin.net/cz8uz1sy6ww01ri6)

**Comment by fmw42** [comment, marked as answer]
ImageMagick uses libtiff to work with tiff files.  But it does not support old style JPG compression in the tiff. It is asking you to use some other tool to change the JPG compression to a newer style JPG compression in the TIFF file, otherwise, ImageMagick will not be able to open the file.

**Comment by dlemstra** [comment]
This is not an error message but a warning for the company that created this tiff file. You can suppress that warning with the `-quiet` option:
`magick -quiet gravesonly-2.tif 001.TIF -compose over -composite test.tif`.

**Comment by P2AK** [comment]
Thanks for the explanation dlemstra and  fmw42.   The issue ended up being due to my failure to specify the `[0]` channel for the TIF files.  

I was able to get the results I was hoping for, without the JPEG compression warning, using this modified command:
`convert 001.TIF[0] gravesonly-2.tif[0] -compose over -composite test.tif`

My TIF files were also out of order in my in initial command, but that wasn't producing the warning message.

The `-quiet` flag does suppress the warning when I don't specify the channel for the TIF files, but I was only able to get the results I was looking for after I used the `[0]` channel suffix for each TIF file.
